azext_iot/tests/iothub/configurations/test_iot_config_unit.py

```diff
@@ -899,3 +899,40 @@ def test_config_apply_edge_error(
                 hub_name=mock_target["entity"],
                 content=sample_config_edge_malformed,
             )
+
+
+class TestConfigTestQueries:
+    @pytest.fixture(params=[200])
+    def serviceclient(self, mocker, fixture_ghcs, fixture_sas, request):
+        service_client = mocker.patch(path_service_client)
+        service_client.return_value = build_mock_response(mocker, request.param, {})
+        return service_client
+
+    def test_config_queries(self, fixture_cmd, serviceclient):
+        subject.iot_hub_configuration_test_queries(
+            cmd=fixture_cmd,
+            hub_name=mock_target["entity"],
+            target_condition="tags.building=43 and tags.environment='test'",
```

Comment:
you can keep the target_condition and custom metric queries as parameterized - 

I thought of some extra tests:
have some failure cases (you want to hit both if statements so you will need at least two)
have some that are multiple queries that pass
have some multiple queries that are a mixed of fail/pass

Reply:
test update!